device/cuda/src/seeding/seed_finding.cu

```diff
@@ -195,19 +195,22 @@ seed_finding::output_type seed_finding::operator()(
     CUDA_ERROR_CHECK(cudaGetLastError());
 
     // Get the summary values.
-    device::seeding_global_counter globalCounter_host;
-    CUDA_ERROR_CHECK(cudaMemcpyAsync(&globalCounter_host,
+    vecmem::unique_alloc_ptr<device::seeding_global_counter>
+        globalCounter_host =
+            vecmem::make_unique_alloc<device::seeding_global_counter>(
+                *(m_mr.host));
```

Comment:
:thinking: We currently use the algorithms with separate host and device memory resources. But it would be nice to keep the ability of using a single `vecmem::cuda::managed_memory_resource`. Mainly for testing purposes.

While a shared memory resource in general would require quite a few code changes (since explicit copies are not needed when using that sort of a thing), at least for now we should check whether `m_mr.host` is a `nullptr` or not. We can think about taking a different code path with different memory resource setups in the future.